File: app/src/main/java/com/fivespecial/ploking/Maps/Calculation.java
package com.fivespecial.ploking.maps;
import android.location.Location;

import java.util.List;

public class Calculation {

    BinLocation binLocation;
    double dLat, dLong;
//    List<BinLocation> nearBin;

    public Calculation() {
    }

    public int NearBins(List<BinLocation> binLocationList, double currentLat, double currentLong) {
        int count = 0;

        Location locationB = new Location("point B");
        locationB.setLatitude(currentLat);
        locationB.setLongitude(currentLong);

        for(int i = 0; i < binLocationList.size(); i++) {
            binLocation = binLocationList.get(i);

            if(binLocation.getLatitude() != null && binLocation.getLongitude() != null) {
                dLat = binLocation.getLatitude();
                dLong = binLocation.getLongitude();
            }

            Location locationA = new Location("point A");
            locationA.setLatitude(dLat);
            locationA.setLongitude(dLong);

            if(locationA.distanceTo(locationB) < 1000){
//                nearBin.add(binLocation);
                count++;
            }
        }
            return count;
    }



}

File: app/src/main/java/com/fivespecial/ploking/Maps/Calculation.kt
package com.fivespecial.ploking.maps

import android.location.Location

class Calculation {

    //private val nearBin = ArrayList<BinLocation>()

    fun nearBins(binLocationList: List<BinLocation>, currentLat: Double, currentLong: Double): Int {

        var count = 0

        binLocationList.forEach { bin ->

            if(bin.latitude != null && bin.longitude != null) {

                // Point A
                val locationA: Location = Location("point A").apply {
                    latitude = bin.latitude!!
                    longitude = bin.longitude!!
                }

                // Point B
                val locationB = Location("point B").apply {
                    latitude = currentLat
                    longitude = currentLong
                }

                if(locationA.distanceTo(locationB) < 1000) {
                    // nearBin.add(bin)

                    count++
                }
            }
        }

        return count
    }

}
